src/lib/interrupt.c

```diff
@@ -215,7 +215,7 @@ static uint32_t irq_enable_child(struct irq_desc *parent, int irq,
 
 	/* enable the parent interrupt */
 	if (parent->enabled_count == 0)
-		arch_interrupt_enable_mask(1 << SOF_IRQ_NUMBER(irq));
+		arch_interrupt_enable_mask(1 << parent->irq);
```

Comment:
this looks similar to a previous patch and could be merged.

Reply:
You mean patch "interrupt: SOF_IRQ_NUMBER() for DSP interrupts is not needed?" Well, they both get rid of SOF_IRQ_NUMBER(), that's true, but in different ways and for different reasons, so, I think it's clearer if they're kept apart.